service/history/replication/executable_backfill_history_events_task.go

```diff
@@ -0,0 +1,322 @@
+			tag.WorkflowNamespaceID(e.NamespaceID),
+			tag.WorkflowID(e.WorkflowID),
+			tag.WorkflowRunID(e.RunID),
+			tag.TaskID(e.ExecutableTask.TaskID()),
+			tag.Error(err),
+		)
+		return err
+	}
+}
+
+func (e *ExecutableBackfillHistoryEventsTask) MarkPoisonPill() error {
+	if e.markPoisonPillAttempts >= MarkPoisonPillMaxAttempts {
+		taskInfo := &persistencespb.ReplicationTaskInfo{
+			NamespaceId:         e.NamespaceID,
+			WorkflowId:          e.WorkflowID,
+			RunId:               e.RunID,
+			TaskId:              e.ExecutableTask.TaskID(),
+			TaskType:            enumsspb.TASK_TYPE_REPLICATION_SYNC_VERSIONED_TRANSITION,
+			VisibilityTime:      timestamppb.New(e.TaskCreationTime()),
+			NewRunId:            e.taskAttr.NewRunId,
+			VersionedTransition: e.taskAttr.VersionedTransition,
+		}
+		events, newRunEvents, err := e.getDeserializedEvents()
+		if err != nil {
+			taskInfo.FirstEventId = -1
+			taskInfo.NextEventId = -1
+			taskInfo.Version = -1
+			e.Logger.Error("MarkPoisonPill reached max attempts, deserialize failed",
+				tag.SourceCluster(e.SourceClusterName()),
+				tag.ReplicationTask(taskInfo),
+				tag.Error(err),
+			)
+			return nil
+		}
+		taskInfo.FirstEventId = events[0][0].GetEventId()
+		taskInfo.NextEventId = events[len(events)-1][len(events[len(events)-1])-1].GetEventId() + 1
+		taskInfo.Version = events[0][0].GetVersion()
+		e.Logger.Error("MarkPoisonPill reached max attempts",
+			tag.SourceCluster(e.SourceClusterName()),
+			tag.ReplicationTask(taskInfo),
+			tag.NewAnyTag("NewRunFirstEventId", newRunEvents[0].GetEventId()),
+			tag.NewAnyTag("NewRunLastEventId", newRunEvents[len(newRunEvents)-1].GetEventId()),
+		)
+		return nil
+	}
+	e.markPoisonPillAttempts++
+
+	shardContext, err := e.ShardController.GetShardByNamespaceWorkflow(
+		namespace.ID(e.NamespaceID),
+		e.WorkflowID,
+	)
+	if err != nil {
+		return err
+	}
+
+	eventBatches := [][]*historypb.HistoryEvent{}
+	for _, eventsBlob := range e.taskAttr.EventBatches {
+		events, err := e.EventSerializer.DeserializeEvents(eventsBlob)
+		if err != nil {
+			e.Logger.Error("unable to enqueue sync versioned transition replication task to DLQ, ser/de error",
+				tag.ShardID(shardContext.GetShardID()),
+				tag.WorkflowNamespaceID(e.NamespaceID),
+				tag.WorkflowID(e.WorkflowID),
+				tag.WorkflowRunID(e.RunID),
+				tag.TaskID(e.ExecutableTask.TaskID()),
+				tag.Error(err),
+			)
+			return nil
+		} else if len(events) == 0 {
+			e.Logger.Error("unable to enqueue sync versioned transition replication task to DLQ, no events",
+				tag.ShardID(shardContext.GetShardID()),
+				tag.WorkflowNamespaceID(e.NamespaceID),
+				tag.WorkflowID(e.WorkflowID),
+				tag.WorkflowRunID(e.RunID),
+				tag.TaskID(e.ExecutableTask.TaskID()),
+			)
+			return nil
+		}
+		eventBatches = append(eventBatches, events)
+	}
+
+	// TODO: GetShardID will break GetDLQReplicationMessages we need to handle DLQ for cross shard replication.
+	taskInfo := &persistencespb.ReplicationTaskInfo{
```

Comment:
nit: probably easier if we just add raw task info to the replicationTask message.

Reply:
added raw_task_info to message ReplicationTask